chimney/src/main/scala/io/scalaland/chimney/DerivedTransformer.scala

```diff
@@ -0,0 +1,41 @@
+package io.scalaland.chimney
+
+import shapeless._
+import shapeless.labelled._
+import shapeless.tag._
+
+
+trait DerivedTransformer[From, To, Modifiers <: HList] {
+
+  def transform(src: From, modifiers: Modifiers): To
+}
+
+object DerivedTransformer {
+
+  implicit def fromNonDerived[T, U, Modifiers <: HList]
+    (implicit transformer: Transformer[T, U]): DerivedTransformer[T, U, Modifiers] =
+    (obj: T, _: Modifiers) =>
+      transformer.transform(obj)
+
+  implicit def identityTransformer[T, Modifiers <: HList]: DerivedTransformer[T, T, Modifiers] =
+    (obj: T, _: Modifiers) => obj
+
+  implicit def hnilCase[From, FromG <: HList, Modifiers <: HList]: DerivedTransformer[FromG @@ From, HNil, Modifiers] =
+    (_: FromG @@ From, _: Modifiers) => HNil
+
+  implicit def hconsCase[From, FromG <: HList, Label <: Symbol, ToFieldT, TailToG <: HList, Modifiers <: HList]
+    (implicit vp: ValueProvider[FromG, From, ToFieldT, Label, Modifiers],
+     tailTransformer: DerivedTransformer[FromG @@ From, TailToG, Modifiers])
+  : DerivedTransformer[FromG @@ From, FieldType[Label, ToFieldT] :: TailToG, Modifiers] = {
+    (obj: FromG @@ From, modifiers: Modifiers) =>
+      field[Label](vp.provide(obj, modifiers)) :: tailTransformer.transform(obj, modifiers)
+  }
+
+  implicit def gen[From, To, FromG, ToG, Modifiers <: HList]
+    (implicit fromLG: LabelledGeneric.Aux[From, FromG],
+     toLG: LabelledGeneric.Aux[To, ToG],
+     genTransformer: DerivedTransformer[FromG @@ From, ToG, Modifiers]): DerivedTransformer[From, To, Modifiers] = {
+    (src: From, modifiers: Modifiers) =>
+      toLG.from(genTransformer.transform(tag[From](fromLG.to(src)), modifiers))
```

Comment:
We should have some consistent naming, e.g.:

 * `from` is `From`,
 * `fromG` is `FromG`.

Right now `FromG` type has `from` value, while `From` has `src` and `obj`.